src/contracts/atlas/AtlETH.sol

```diff
@@ -96,29 +117,54 @@ abstract contract AtlETH is Permit69 {
         _balanceOf[address(0xa7145)].balance += uint128(amount);
     }
 
-    function _withdrawAccounting(uint256 amount, address spender) internal {
+    function _bond(address account, uint256 amount) internal {
+        EscrowAccountData memory accountData = _balanceOf[account];
+        if (accountData.balance - accountData.holds < amount) {
+            revert InsufficientUnbondedBalance({ balance: accountData.balance - accountData.holds, requested: amount });
+        }
+        accountData.holds += uint128(amount);
+        _balanceOf[account] = accountData;
+    }
+
+    // Returns the allowed withdrawal amount which may be <= the requested amount param
+    function _withdrawAccounting(uint256 amount, address spender) internal returns (uint256) {
         EscrowNonce memory nonceData = nonces[spender];
         EscrowAccountData memory accountData = _balanceOf[spender];
 
-        uint128 _amount = uint128(amount);
-
-        if (nonceData.withdrawalAmount < _amount) {
-            revert InsufficientRedeemedBalance(uint256(nonceData.withdrawalAmount), amount);
+        if (nonceData.withdrawalAmount < amount) {
+            revert InsufficientWithdrawableBalance({ balance: nonceData.withdrawalAmount, requested: amount });
         }
```

Comment:
This is reverting if trying to withdraw more than the unbonding balance.
Shouldn't we consider the free balance as well (non-bonded nor unbonding)?
Doesn't a holder that never bonded its athEth would revert here? Although having enough "free" balance?

Reply:
Added a test for that case as well - user does not ever bond or unbond, should still be able to withdraw full amount. Think this PR needs more tests to ensure we're handling the edge cases.